### ISSUE TITLE
Additional requests when there're no results with dynamic widgets

### ISSUE THREAD

**Issue by dziugas-liaudinskas** [typesense/typesense-instantsearch-adapter#173]
## Description
When using DynamicWidgets and the price range filter simultaneously, an issue arises when encountering scenarios with no search results. In these cases, multiple redundant search requests are triggered, and the received results do not reflect the applied filters. It is expected that a "no results" screen would be displayed, similar to the behavior observed when using Algolia.

### Example with Typesense adapter
1. Clone [this](https://github.com/dziugas-liaudinskas/typesense-dynamic-filters-demo) repo
2. Run `npm install && npm run build && npm start` and open localhost
3. Select the following filters:
- "rating" => "5"
- "free_shipping" => "true"
- "brand" => "Spigen"
4. Move the price filter "from" to "23$"
5. Move the price filter "to" to "28$" so that there would not be any products meeting the filters
6. You can see screen flickers and selected filters get removed. In the network tab, multiple requests can be seen

### Example with Algolia

1. Clone [this](https://github.com/dziugas-liaudinskas/algolia-example) repo
2. Run `npm i && npm run dev`, open localhost and go to `/category/speakers`
3. Select the following filters:
- "brand" => "BIC America"
- "rating" => "5"
4. Move the price filter "from" to "115$"
5. Move the price filter "to" to "130$" so that there would not be any products meeting the filters
6. You can see that there are no results but filters remain as you selected

## Expected Behavior
When there are no available results there should not be additional requests with filters removed. The user should see the "no results" screen

## Actual Behavior
When then are no available results screen flickers and selected filters get removed. In the network tab, multiple requests can be seen

## Metadata
"typesense-instantsearch-adapter": "^2.7.0"

**Comment by jasonbosco** [comment]
It looks like the Typesense example [uses `react-instantsearch-hooks-web`](https://github.com/dziugas-liaudinskas/typesense-dynamic-filters-demo/blob/d1b818376c60c14511786cc45c2526bb3dbbd8ed/src/App.tsx#L7) whereas the Algolia example [uses `instantsearch.js`](https://github.com/dziugas-liaudinskas/algolia-example/blob/0b5ccbda84df684f787d521ab73af2ee826dc58d/app/components/Search.tsx#L2). 

I suspect the issue is most likely related to `react-instantsearch-hooks-web` and not with the adapter (just based on anecdotal reports I've heard from other react-instantsearch-hooks users about unrelated issues). 

For eg, with the same dataset that your Typesense example is using, [this demo app](https://ecommerce-store.typesense.org/?products%5BrefinementList%5D%5Bbrand%5D%5B0%5D=Spigen&products%5BhierarchicalMenu%5D%5Bcategories.lvl0%5D%5B0%5D=Cell%20Phones&products%5Btoggle%5D%5Bfree_shipping%5D=true&products%5Brange%5D%5Bprice%5D=23%3A24&products%5BratingMenu%5D%5Brating%5D=4) uses `instantsearch.js` and doesn't have the same issue, with the same reproduction steps. 

Although, I'm not using `dynamicWidgets` in that example. Could you try using `dynamicWidgets` with the example I shared above and see if you can reproduce the issue?

**Comment by dziugas-liaudinskas** [comment]
Also, the Algolia example above also uses [dynamic widgets from `react-instant-search-hooks-web`](https://github.com/dziugas-liaudinskas/algolia-example/blob/0b5ccbda84df684f787d521ab73af2ee826dc58d/app/components/Filters.tsx#L3)

**Comment by dziugas-liaudinskas** [comment]
@jasonbosco Would you mind taking a look at this when you have a moment? I would greatly appreciate your input.

**Comment by jasonbosco** [comment]
@dziugas-liaudinskas Sorry about the delay, and thank you for the detailed reproduction steps. That was super helpful. 

I was able to identify where the different is coming from. I've pushed out a fix in `v2.7.1-0` of the adapter. Could you give it a shot now?

**Comment by dziugas-liaudinskas** [comment]
@jasonbosco As I can see issue still persists. I pushed the new branch `update-2.7.1-0` to the [example that I provided in the description of this issue](https://github.com/dziugas-liaudinskas/typesense-dynamic-filters-demo/tree/update-2.7.1-0). There "typesense-instantsearch-adapter" is updated to `v2.7.1-0` but seems like nothing changed.

**Comment by jasonbosco** [comment]
@dziugas-liaudinskas I'm not too familiar with react-instantsearch-hooks, but it seems like the fix worked in this other [instantsearch.js repo](https://github.com/dziugas-liaudinskas/instant-search-js-dynamic) you shared, once you upgrade to `v2.7.1-0`. Could you verify this?

**Comment by dziugas-liaudinskas** [comment]
@jasonbosco  The issue still persists on that repo too with an updated version. I pushed a [new branch with an updated version (branch name: update-v2.7.1-0)](https://github.com/dziugas-liaudinskas/instant-search-js-dynamic/tree/update-v2.7.1-0) if you are keen to try it. Maybe we do something differently? Added screen recording to check if we follow the same steps.


https://github.com/typesense/typesense-instantsearch-adapter/assets/37068409/cd9eac9c-09dc-4e8d-9da5-1bd515d39389
